modules/react-native-status/ios/RCTStatus/Jail.m

```diff
@@ -0,0 +1,234 @@
+#import "RCTStatus.h"
+
+//source: http://stackoverflow.com/a/23387659/828487
+#define NSStringMultiline(...) [[NSString alloc] initWithCString:#__VA_ARGS__ encoding:NSUTF8StringEncoding]
+
+@implementation HandlersJs
+
+- (void)log:(JSValue *)data
+{
+    NSLog(@"jail log: %@", [data toString]);
+}
+
+- (NSString *)send:(JSValue *)payload
+{
+    char * result = CallRPC((char *) [[payload toString] UTF8String]);
+    
+    return [NSString stringWithUTF8String: result];
+}
+
+- (void)sendAsync:(JSValue *)args
+{
+    // TODO(rasom): fix this black magic, need to figure how to pass more than one
+    // parameter to sendAsync
+    JSValue *payload = [args callWithArguments:@[@0]];
+    JSValue *callback = [args callWithArguments:@[@1]];
+    
+    dispatch_async( dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        NSString *result = [self send:payload];
+        dispatch_async( dispatch_get_main_queue(), ^{
+            [callback callWithArguments:@[result]];
+        });
+    });
+}
+
+- (BOOL)isConnected
+{
+    return YES;
+}
+
+- (void)sendSignal:(JSValue *)data
+{
+    [Status jailEvent:_chatId data:[data toString]];
+}
+
+- (void)setCahtId:(NSString *)chatId
+{
+    _chatId = chatId;
+}
+
+- (void)addToContext:(JSContext *)context
+{
+    [context setObject:self forKeyedSubscript:@"statusNativeHandlers"];
+}
+
+@end
+
+@implementation Cell
+@end
+
+@implementation Jail
+
+- (void)initJail:(NSString *)js
+{
+    _initialJs = js;
+}
+
+- (NSDictionary *)parseJail:(NSString *)chatId
+                   withCode:(NSString *)js
+{
+    Cell *cell = [self createCell:chatId withCode:js];
+    JSContext *context = cell.context;
+    [_cells setValue:cell forKey:chatId];
+    
+    JSValue *catalog = context[@"catalog"];;
+    JSValue *exception = [context exception];
+    NSString *error;
+    if(exception != nil) {
+        error = [exception toString];
+        [context setException:nil];
+    }
+    NSDictionary *result = [[NSDictionary alloc] initWithObjectsAndKeys:[catalog toString], @"result", error, @"error", nil];
+    
+    return result;
+}
+
+- (NSDictionary *)call:(NSString *)chatId
+                  path:(NSString *)path
+                params:(NSString *)params
+{
+    Cell *cell = [_cells valueForKey:chatId];
+    JSContext *context = cell.context;
+    if(cell == nil) {
+        return [[NSDictionary alloc] initWithObjectsAndKeys:nil, @"result", @"jail is not initialized", @"error", nil];
+    }
+    JSValue *callResult = [context[@"call"] callWithArguments:@[path, params]];
+    JSValue *exception = [context exception];
+    NSString *error;
+    if(exception != nil) {
+        error = [exception toString];
+        [context setException:nil];
+    }
+    NSDictionary *result = [[NSDictionary alloc] initWithObjectsAndKeys:[callResult toString], @"result", error, @"error", nil];
+    
+    return result;
+}
+
+- (Cell *)createCell:(NSString *)chatId
+                 withCode:(NSString *)js
+{
```

Comment:
How come we are using this convention here? Same question for rest of file. I thought we did:

```
foobar {
   ...
}
```

Reply:
for no reason, probably i just started doing this at some point, will fix this